### PR TITLE
Add gatsby-compression-plugin.

### PR DESCRIPTION
**Pull request by hbowden**
A plugin for gzipping assets.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Nice! Should this also work on non-webpack handled assets e.g. images generated by gatsby-plugin-sharp? Perhaps go through files and see which ones don't have gz and gzip them too?

File: packages/gatsby-compression-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  ],
+}
+
+```
+
+That's it, when you build your app you will have gzipped versions of your files.
+
+# Nginx
+
+If your using nginx you can use `gzip_static on;` to serve your gzipped assets, here's a full example.
```
Comment (dustinhorton) [inline packages/gatsby-compression-plugin/README.md]: @hbowden `you’re`

Also, this is rad, nice work.

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit a2f16242f7f2ac7516eb78a946c8863e55ab6418

https://app.netlify.com/sites/using-glamor/deploys/5a0b0379a6188f349996adcb

**Comment by hbowden** [comment]
Thanks for the suggestions, I'll update the pull request. About the failed deploy preview, it seemed to be working before the merge, I'll try to figure out whats broken. Is there a command to run the deploy preview locally for testing? Also I'll look into gzipping the assets created by `gatsby-plugin-sharp` and dynamic imports which are not currently supported https://github.com/webpack-contrib/compression-webpack-plugin/issues/79.

**Comment by KyleAMathews** [comment]
The deploy preview stuff is in bad shape and is being replaced soon so just ignore it for now :-(

**Comment by KyleAMathews** [comment]
@hbowden would love to see this finished up!

**Comment by KyleAMathews** [comment]
Please refile a new PR if you get to it!

**Pull request closed without merging**